### PR TITLE
fast path flip messages in dev conversations to bypass any state updates CORE-10266

### PR DESCRIPTION
**Pull request by mmaxim**
The goal is to make it so we can process these messages faster, to reduce chances of failed flips.

### CODE REVIEW

**Comment by mmaxim** [comment]
cc @maxtaco

File: go/chat/flipmanager.go [hunk 1]
```diff
@@ -616,29 +616,47 @@ func (m *FlipManager) StartFlip(ctx context.Context, uid gregor1.UID, hostConvID
...
+		m.Debug(ctx, "MaybeInjectFlipMessage: failed to update inbox version: %s", err)
+		// charge forward here, we will figure it out
+	}
+	// Unbox the message
+	conv, err := utils.GetUnverifiedConv(ctx, m.G(), uid, convID, types.InboxSourceDataSourceAll)
+	if err != nil {
+		m.Debug(ctx, "MaybeInjectFlipMessage: failed to get conversation for unbox: %s", err)
+		return true
+	}
+	msg, err := NewBoxer(m.G()).UnboxMessage(ctx, boxedMsg, conv.Conv, nil)
```
Comment (joshblum) [inline go/chat/flipmanager.go]: won't these won't get exploded since we don't merge into storage?

File: go/chat/push.go [hunk 2]
```diff
@@ -501,6 +501,13 @@ func (g *PushHandler) Activity(ctx context.Context, m gregor.OutOfBandMessage) (
 				g.Debug(ctx, "chat activity: newMessage: outboxID is empty")
 			}
 
+			// Coin flip manager can completely handle the incoming message
+			if g.G().CoinFlipManager.MaybeInjectFlipMessage(ctx, nm.Message, nm.InboxVers, uid, nm.ConvID,
+				nm.TopicType) {
+				g.Debug(ctx, "chat activity: flip message handled, early out")
+				return
```
Comment (joshblum) [inline go/chat/push.go]: why not allow the rest of the processing so we don't have to worry about weird state with these functions not being called. could return a `*chat1.MessageUnboxed` and have a `ConvSource.PushUnboxed`?

Comment (joshblum) [reply go/chat/push.go]: (or just unbox twice if we're not worried about the perf there)

File: go/chat/flipmanager.go (hunk 1 above)
Comment (mmaxim) [reply go/chat/flipmanager.go]: Why do we care then?

File: go/chat/push.go (hunk 2 above)
Comment (mmaxim) [reply go/chat/push.go]: The point is to skip this processing, I don't think it is necessary for these. The only thing we absolutely need to do is update the inbox version, which this guy does.

**Comment by mmaxim** [comment]
Ran locally

**Pull request merged**